internal/resourcepolicies/resource_policies.go

```diff
@@ -123,15 +136,31 @@ func (p *Policies) match(res *structuredVolume) *Action {
 }
 
 func (p *Policies) GetMatchAction(res any) (*Action, error) {
+	data, ok := res.(VolumeFilterData)
+	if !ok {
+		return nil, errors.New("failed to convert input to VolumeFilterData")
+	}
+
 	volume := &structuredVolume{}
-	switch obj := res.(type) {
-	case *v1.PersistentVolume:
-		volume.parsePV(obj)
-	case *v1.Volume:
-		volume.parsePodVolume(obj)
+	switch {
+	case data.PersistentVolume != nil:
+		volume.parsePV(data.PersistentVolume)
+		if data.PVC != nil {
+			if err := volume.parsePVC(data.PVC); err != nil {
+				return nil, err
+			}
+		}
+	case data.PodVolume != nil:
+		volume.parsePodVolume(data.PodVolume)
+		if data.PVC != nil {
+			if err := volume.parsePVC(data.PVC); err != nil {
```

Comment:
See https://github.com/vmware-tanzu/velero/pull/8713/files#r1970824926, we don't need to check the error.